### PR TITLE
MenuItem Accelerator Property

### LINKED ISSUE #251: Add Accelerator property to MenuItem
MenuItem is missing ``Accelerator`` (which define keyboard shortcuts). It's apparently defined in [Xamarin.Forms.Core.json#L1513](https://github.com/fsprojects/Fabulous/blob/60d2bf7875830b49e52e6d1abe505a9b2a139fe0/tools/Generator/Xamarin.Forms.Core.json#L1513), but it doesn't seem to be correct; there's no "Accelerator" anywhere to be found in any types. I assume it is a simple correction.

### PR DESCRIPTION
**Pull request by SergejDK**
Changed the position of the attached property "Accelerator" so now we can call it:
`View.MenuItem().Accelerator("string for accelerator")`

fix #251

### CODE REVIEW

File: tools/Generator/Xamarin.Forms.Core.json [hunk 1]
```diff
@@ -1660,7 +1660,16 @@
           "name": "Command",
           "defaultValue": "null",
           "inputType": "unit -> unit",
-          "convToModel": "makeCommand"
+          "convToModel": "makeCommand",
+          "attached": [
```
Comment (TimLariviere) [inline tools/Generator/Xamarin.Forms.Core.json]: This `attached` field is meant for a property that is a collection.
It allows to use attached properties on the children of this collection, like Grid.Column is only valid on the direct children of a Grid.
So here, it doesn't make sense to add the Accelerator there, as it's only valid on the MenuItem themselves.

There's 2 options to fix this:
- Add `Accelerator` as an attached property on the `Items` property of `Menu`
- Add `Accelerator` as a standard property on `MenuItem` but with a custom update logic that uses `MenuItem.SetAccelerator(menuItem, value)` instead.

I think the 2nd one is better, even if not quite exactly what XF does.

**Comment by SergejDK** [comment]
Tried to solved it.
Hope it is nearly as expected. 
Looking forward for hints how to improve!

**Comment by TimLariviere** [comment]
This is what I meant, good job! 👍 

The only thing we can improve is the performance by not setting the Accelerator if the previous value is the same as the new one.
We can put add a new case in the JSON file's `updateCode`, but it will be easier to extract it to an actual F# file. (Fabulous/src/Fabulous.Core/ViewConverters.fs serves that purpose)
Take a look at
https://github.com/fsprojects/Fabulous/blob/892c59bd8e0158e7aedd5e4f8f76f469922b1ec7/src/Fabulous.Core/ViewConverters.fs#L507-L512

And then, instead of writing the F# code into the json, you just point it to the function in ViewConverters.fs
https://github.com/fsprojects/Fabulous/blob/892c59bd8e0158e7aedd5e4f8f76f469922b1ec7/tools/Generator/Xamarin.Forms.Core.json#L102-L109

File: src/Fabulous.Core/ViewConverters.fs [hunk 2]
```diff
@@ -638,6 +638,14 @@ module Converters =
             NavigationPage.SetTitleView(target, null)
         | _, _ -> ()
 
+    /// Update the AcceleratorProperty of a MenuItem, given previous and current Accelerator
+    let internal updateAccelerator prevValue currValue (target: Xamarin.Forms.MenuItem) =
+        match prevValue, currValue with
+        | ValueNone, ValueNone -> ()
+        | ValueSome prevVal, ValueSome newVal when prevVal = newVal -> ()
+        | _, ValueNone -> ()
```
[the remark this answers is not in the document]
Comment (TimLariviere) [reply src/Fabulous.Core/ViewConverters.fs]: If the default value of `Accelerator` is null, then yes. :)

**Comment by TimLariviere** [comment]
Thanks!
I will merge it once the build is done.

**Comment by SergejDK** [comment]
great! :)

**Pull request merged**